scan.go

```diff
@@ -146,78 +157,243 @@ func ReadRR(q io.Reader, filename string) (RR, error) {
 //              }
 //	}
 //
-// Comments specified after an RR (and on the same line!) are returned too:
+// Comments specified after an RR (and on the same line!) are
+// returned too:
 //
 //	foo. IN A 10.0.0.1 ; this is a comment
 //
-// The text "; this is comment" is returned in Token.Comment. Comments inside the
-// RR are returned concatenated along with the RR. Comments on a line by themselves
-// are discarded.
+// The text "; this is comment" is returned in Token.Comment.
+// Comments inside the RR are returned concatenated along with the
+// RR. Comments on a line by themselves are discarded.
+//
+// To prevent memory leaks it is important to always fully drain the
+// returned channel. If an error occurs, it will always be the last
+// Token sent on the channel.
```

Comment:
Might be nice to put in a little example code.

Reply:
I can do that, I avoided it because the `defer` block (like in the coredns PR) doesn’t `gofmt` that nicely and is a bit messy. I had thought about adding a `func DrainTokenChan(chan *Token)` that callers could `defer`.